File: packages/models/lib/src/messages/messages_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'package:models/src/messages/messages_model.dart';

part 'messages_response.freezed.dart';

part 'messages_response.g.dart';

@freezed
class MessagesResponse with _$MessagesResponse {
  const factory MessagesResponse({
    required List<MessagesModel> messagesList,
    required int currentPage,
    required int totalItems,
    required int totalPages,
    required int itemsPerPage,
  }) = _MessagesResponse;

  factory MessagesResponse.fromJson(Map<String, dynamic> json) =>
      _$MessagesResponseFromJson(json);
}

File: java_server/src/main/java/xyz/qruto/java_server/models/responses/MessagesResponse.java
package xyz.qruto.java_server.models.responses;

import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class MessagesResponse {
    private List<MessageBriefResponse> messagesList;
    private int currentPage;
    private int totalItems;
    private int totalPages;
    private int itemsPerPage;
}
